namecheap/resource_namecheap_record.go

```diff
@@ -79,14 +83,33 @@ func resourceNameCheapRecordCreate(d *schema.ResourceData, meta interface{}) err
 		TTL:        d.Get("ttl").(int),
 	}
 
-	_, err := client.AddRecord(d.Get("domain").(string), &record)
-
-	if err != nil {
-		mutex.Unlock()
-		return fmt.Errorf("Failed to create namecheap Record: %s", err)
+	apiThrottleBackoffTime := 2
+
+	for {
+		_, err := client.AddRecord(d.Get("domain").(string), &record)
+		if err != nil {
+			log.Printf("[TRACE] Err: %v", err.Error())
+
+			if strings.Contains(err.Error(), "expected element type <ApiResponse> but have <html>") {
+				log.Printf("[WARN] Bad Namecheap API response received, backing off for %d seconds...",
+					apiThrottleBackoffTime)
+				time.Sleep(time.Duration(apiThrottleBackoffTime) * time.Second)
+				apiThrottleBackoffTime = apiThrottleBackoffTime * ncBackoffMultiplier
+				if apiThrottleBackoffTime > ncMaxThrottleRetry {
+					log.Printf("[ERROR] API Retry Limit Reached. Couldn't find namecheap record: %v", err)
+					break
```

Comment:
I think we'd need to unlock in this case and the `continue` below.

Reply:
Are you sure about that? I think since it Locks right in the start of these methods, we really only need to unlock on a `return`. Thoughts? 

Also double check the delete method, it locks then unlocks immediately. Maybe I'm not understanding what these mutexes are doing... I'm familiar with resource locking and all that but perhaps I'm missing something here.

Reply:
You're right about not needed an unlock on continue/break. My bad, hah! 

On master the delete calls look to unlock properly, do you see something different?